core/PRP/Geometry/plDrawableSpans.cpp

```diff
@@ -661,14 +661,12 @@ void plDrawableSpans::clearSpans()
     fSpanSourceIndices.clear();
 }
 
-size_t plDrawableSpans::addIcicle(const plIcicle& span)
+size_t plDrawableSpans::addIcicle(plIcicle* span)
 {
-    plIcicle* iceCopy = new plIcicle(span);
-    fIcicles.push_back(iceCopy);
-    size_t iceId = fIcicles.size() - 1;
-    fSpans.push_back(fIcicles[iceId]);
-    fSpanSourceIndices.push_back(iceId | kSpanTypeIcicle);
-    return fSpans.size() - 1;
+    fIcicles.push_back(span);
+    fSpans.push_back(span);
+    fSpanSourceIndices.push_back((fIcicles.size() - 1) | kSpanTypeIcicle);
+    return fIcicles.size() - 1;
```

Comment:
And now you've changed the other part (the return value)...  Hmm, I really don't remember how all this works, nor why we keep two separate vectors.  Who is using these values anyway?

Reply:
I suspect the reason there are two vectors is because we match what Plasma does. It used to have many different kinds of not necessarily icicle spans. It's all mostly dead end now though... For the sake of IO, we really only care about the icicle spans.